packages/babel-plugin-transform-es2015-parameters/test/fixtures/parameters/regression-5787/expected.js

```diff
@@ -0,0 +1,8 @@
+function f(a) {
+  var _ref, _ref2, _ref3;
+
+  var b = (_ref = (arguments.length <= 1 ? 0 : arguments.length - 1) - 3 + 1, arguments.length <= _ref || _ref < 1 ? undefined : arguments[_ref]);
```

Comment:
It's actually hairier than I thought. The code we see is a result of 2 generators working independently: index-getter and length-getter. The length-getter includes that check so that `someRestParam.length` always returns the correct non-negative value and it is only a coincidence that the check is irrelevant in the specific case we saw when the expressions generated by index-getter and length-getter are combined. The code will fail in cases that depending on the invariant of `arr.length` being non-negative. For example:

```js
let x
let b = rest[x = rest.length - 3];
// => x might have incorrect value because rest.length could be negative
```
I propose to keep that check there to avoid introducing complexity just to have a minor optimization to a specific code path. Please let me what you think.